adafruit_midi/__init__.py

```diff
@@ -55,39 +58,100 @@ class MIDI:
     PITCH_BEND = 0xE0
     CONTROL_CHANGE = 0xB0
 
-    def __init__(self, midi_in=usb_midi.ports[0], midi_out=usb_midi.ports[1], *, in_channel=None,
-                 out_channel=0, debug=False):
+    def __init__(self, midi_in=usb_midi.ports[0], midi_out=usb_midi.ports[1], *,
+                 in_channel=None, out_channel=0, debug=False, in_buf_size=30):
         self._midi_in = midi_in
         self._midi_out = midi_out
-        self._in_channel = in_channel
-        self._out_channel = out_channel
+        self.in_channel = in_channel
+        self.out_channel = out_channel
         self._debug = debug
+        # This input buffer holds what has been read from midi_in
+        self._in_buf = bytearray(0)
+        self._in_buf_size = in_buf_size
         self._outbuf = bytearray(4)
+        self._skipped_bytes = 0
 
     @property
     def in_channel(self):
         """The incoming MIDI channel. Must be 0-15. Correlates to MIDI channels 1-16, e.g.
-        ``in_channel(3)`` will listen on MIDI channel 4. Default is 0."""
+        ``in_channel(3)`` will listen on MIDI channel 4.
+        Can also listen on multiple channels, e.g. ``in_channel((0,1,2))``
+        will listen on MIDI channels 1-3 or ``in_channel("ALL")`` for every channel.
+        Default is None."""
         return self._in_channel
 
+    # pylint: disable=attribute-defined-outside-init
```

Comment:
I've seen that and asked about it possibly in the dotstar libraries. I did it this way to avoid superfluous code as I've been getting really tight on memory with my application on CPX board.